### PR TITLE
More Nondet Monad improvements

### PR DESCRIPTION
**Pull request by corlewis**
More improvements to Nondet Monad, following on from https://github.com/seL4/l4v/pull/665. In general this improves style and some proofs, removes duplicate lemmas, and stays in sync with coming changes to Trace Monad.

### CODE REVIEW

File: lib/Monads/nondet/Nondet_VCG.thy [hunk 1]
```diff
@@ -6,16 +6,17 @@
  *)
 
 theory Nondet_VCG
-imports
-  Nondet_Lemmas
-  WPSimp
+  imports
```
Comment (Xaphiosis) [inline lib/Monads/nondet/Nondet_VCG.thy:9]: when did we start indenting imports? I know the autoindenter likes to do it, but I'm not sure that that's our policy. @lsf37 ?

**Review by Xaphiosis — APPROVED** [review]
Other than the `import` indentation question, thumbs up from me.

File: lib/Monads/nondet/Nondet_VCG.thy (hunk 1 above)
Comment (corlewis) [reply lib/Monads/nondet/Nondet_VCG.thy:9]: I noticed that most of the files in `nondet` already had indented imports so I decided to go with the autoindenter and make them consistent. I don't know if we have an official policy, but I'd also be happy to go in the opposite direction and make them consistent with the rest of the repository.

Comment (lsf37) [reply lib/Monads/nondet/Nondet_VCG.thy:9]: We probably shouldn't fight the auto indenter on this. We used to not indent them, but I'd be happy with indented import statements (and I think I have done that a few times myself now)

**Pull request merged**